Userland/Libraries/LibWeb/HTML/History.cpp

```diff
@@ -44,14 +44,71 @@ WebIDL::ExceptionOr<void> History::replace_state(JS::Value data, String const&,
     return shared_history_push_replace_state(data, url, IsPush::No);
 }
 
+// https://html.spec.whatwg.org/multipage/history.html#dom-history-length
+WebIDL::ExceptionOr<u64> History::length() const
+{
+    // 1. If this's associated Document is not fully active, then throw a "SecurityError" DOMException.
+    if (!m_associated_document->is_fully_active())
+        return WebIDL::SecurityError::create(realm(), "Cannot perform length on a document that isn't fully active."sv);
+
+    // 2. Return the number of entries in the top-level browsing context's joint session history.
+    auto const* browsing_context = m_associated_document->browsing_context();
+    return browsing_context->session_history().size();
+}
+
+// https://html.spec.whatwg.org/multipage/history.html#dom-history-go
+WebIDL::ExceptionOr<void> History::go(long delta = 0)
+{
+    // 1. Let document be this's associated Document.
+
+    // 2. If document is not fully active, then throw a "SecurityError" DOMException.
+    if (!m_associated_document->is_fully_active())
+        return WebIDL::SecurityError::create(realm(), "Cannot perform go on a document that isn't fully active."sv);
+
+    // 3. If delta is 0, then act as if the location.reload() method was called, and return.
+    // 4. Traverse the history by a delta with delta and document's browsing context.
+    auto* browsing_context = m_associated_document->browsing_context();
+    auto current_entry_index = browsing_context->session_history_index();
+    auto next_entry_index = current_entry_index + delta;
+    auto sessions = browsing_context->session_history();
```

Comment:
```suggestion
    auto const& sessions = browsing_context->session_history();
```

No need to make a copy